Fuji node on Virtual Machine gives panic: counter cannot decrease in value

**Describe the bug**
I'm trying to run a fuji node on a Virtualbox VM with ubuntu 22.04 on it. Once i've installed golang and github i build avalanchego and when i run it i get this error:

```
panic: counter cannot decrease in value
```

**To Reproduce**
I'm working with virtual box 7 with a 22.04 ubuntu distro. Once the virtual machine is set up you install go and git. Once you're done you can clone the avalanchego repository and build it using /scripts/build.sh

Once this is done starting the node will produce the error above

**Logs**

```
goroutine 3064 [running]:
github.com/prometheus/client_golang/prometheus.(counter).Add(0xc006587020?, 0xc006a57900?)
 /home/vboxuser/go/pkg/mod/github.com/prometheus/client_golang@v1.14.0/prometheus/counter.go:109 +0xe9
github.com/ava-labs/avalanchego/database/meterdb.(Database).Put(0xc002ecb0e0, {0xc006a57900, 0x8c, 0xff}, {0xc00f60728e, 0x1, 0x1})
 /home/vboxuser/Avalanche/avalanchego/database/meterdb/db.go:167 +0x132
github.com/ava-labs/avalanchego/database/meterdb.(Database).Put(0xc0065870b0, {0xc006a57900, 0x8c, 0xff}, {0xc00f60728e, 0x1, 0x1})
 /home/vboxuser/Avalanche/avalanchego/database/meterdb/db.go:163 +0xa4
github.com/ava-labs/avalanchego/database/prefixdb.(Database).Put(0xc0057fcd20, {0xc008de4780?, 0x4c?, 0x47?}, {0xc00f60728e, 0x1, 0x1})
 /home/vboxuser/Avalanche/avalanchego/database/prefixdb/db.go:138 +0x1b0
github.com/ava-labs/avalanchego/database/prefixdb.(Database).Put(0xc005ea5730, {0xc012a00550?, 0x50?, 0x76fcc6e4e5b8?}, {0xc00f60728e, 0x1, 0x1})
 /home/vboxuser/Avalanche/avalanchego/database/prefixdb/db.go:138 +0x1b0
github.com/ava-labs/coreth/core/rawdb.WriteSyncStorageTrie({0x76fc6e6e57a8, 0xc0065c6280}, {0xd9, 0x2a, 0x1c, 0xac, 0x66, 0x24, 0x48, 0xa5, ...}, ...)
 /home/vboxuser/go/pkg/mod/github.com/ava-labs/coreth@v0.13.4-0.20240506124912-82b6c4e91557/core/rawdb/accessors_state_sync.go:124 +0x20d
github.com/ava-labs/coreth/sync/statesync.(trieQueue).RegisterStorageTrie(...)
 /home/vboxuser/go/pkg/mod/github.com/ava-labs/coreth@v0.13.4-0.20240506124912-82b6c4e91557/sync/statesync/trie_queue.go:51
github.com/ava-labs/coreth/sync/statesync.(mainTrieTask).OnLeafs(0xc00334d5f0, {0x76fc6e7fa580, 0xc0091f45d0}, {0xc00dbbb508?, 0x400, 0x46f?}, {0xc01260c008, 0x400, 0xb?})
 /home/vboxuser/go/pkg/mod/github.com/ava-labs/coreth@v0.13.4-0.20240506124912-82b6c4e91557/sync/statesync/trie_sync_tasks.go:77 +0x645
github.com/ava-labs/coreth/sync/statesync.(trieSegment).OnLeafs(0xc007f32cb0, {0xc00dbbb508, 0x400, 0x46f?}, {0xc01260c008?, 0x400?, 0x46f?})
 /home/vboxuser/go/pkg/mod/github.com/ava-labs/coreth@v0.13.4-0.20240506124912-82b6c4e91557/sync/statesync/trie_segments.go:364 +0x83
github.com/ava-labs/coreth/sync/client.(CallbackLeafSyncer).syncTask(0xc0046a1b60, {0x2201538, 0xc0031dd040}, {0x2210d18, 0xc007f32cb0})
 /home/vboxuser/go/pkg/mod/github.com/ava-labs/coreth@v0.13.4-0.20240506124912-82b6c4e91557/sync/client/leaf_syncer.go:129 +0x365
github.com/ava-labs/coreth/sync/client.(CallbackLeafSyncer).workerLoop(0xc0046a1b60, {0x2201538, 0xc0031dd040})
 /home/vboxuser/go/pkg/mod/github.com/ava-labs/coreth@v0.13.4-0.20240506124912-82b6c4e91557/sync/client/leaf_syncer.go:71 +0xd0
github.com/ava-labs/coreth/sync/client.(CallbackLeafSyncer).Start.func1()
 /home/vboxuser/go/pkg/mod/github.com/ava-labs/coreth@v0.13.4-0.20240506124912-82b6c4e91557/sync/client/leaf_syncer.go:156 +0x1f
golang.org/x/sync/errgroup.(Group).Go.func1()
 /home/vboxuser/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 3051
 /home/vboxuser/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:75 +0x96
```

--- a/cache/metercacher/metrics.go
+++ b/cache/metercacher/metrics.go
@@ -27,10 +27,10 @@ var (
 
 type metrics struct {
 	getCount *prometheus.CounterVec
-	getTime  *prometheus.CounterVec
+	getTime  *prometheus.GaugeVec
 
 	putCount prometheus.Counter
-	putTime  prometheus.Counter
+	putTime  prometheus.Gauge
 
 	len           prometheus.Gauge
 	portionFilled prometheus.Gauge
@@ -49,8 +49,8 @@ func newMetrics(
 			},
 			resultLabels,
 		),
-		getTime: prometheus.NewCounterVec(
-			prometheus.CounterOpts{
+		getTime: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
 				Namespace: namespace,
 				Name:      "get_time",
 				Help:      "time spent (ns) in get calls",
@@ -62,7 +62,7 @@ func newMetrics(
 			Name:      "put_count",
 			Help:      "number of put calls",
 		}),
-		putTime: prometheus.NewCounter(prometheus.CounterOpts{
+		putTime: prometheus.NewGauge(prometheus.GaugeOpts{
 			Namespace: namespace,
 			Name:      "put_time",
 			Help:      "time spent (ns) in put calls",

--- a/database/leveldb/metrics.go
+++ b/database/leveldb/metrics.go
@@ -19,7 +19,7 @@ type metrics struct {
 	writesDelayedCount prometheus.Counter
 	// total amount of time (in ns) that writes that have been delayed due to
 	// compaction
-	writesDelayedDuration prometheus.Counter
+	writesDelayedDuration prometheus.Gauge
 	// set to 1 if there is currently at least one write that is being delayed
 	// due to compaction
 	writeIsDelayed prometheus.Gauge
@@ -44,7 +44,7 @@ type metrics struct {
 	// size of each level
 	levelSize *prometheus.GaugeVec
 	// amount of time spent compacting each level
-	levelDuration *prometheus.CounterVec
+	levelDuration *prometheus.GaugeVec
 	// amount of bytes read while compacting each level
 	levelReads *prometheus.CounterVec
 	// amount of bytes written while compacting each level
@@ -69,7 +69,7 @@ func newMetrics(namespace string, reg prometheus.Registerer) (metrics, error) {
 			Name:      "writes_delayed",
 			Help:      "number of cumulative writes that have been delayed due to compaction",
 		}),
-		writesDelayedDuration: prometheus.NewCounter(prometheus.CounterOpts{
+		writesDelayedDuration: prometheus.NewGauge(prometheus.GaugeOpts{
 			Namespace: namespace,
 			Name:      "writes_delayed_duration",
 			Help:      "amount of time (in ns) that writes have been delayed due to compaction",
@@ -129,8 +129,8 @@ func newMetrics(namespace string, reg prometheus.Registerer) (metrics, error) {
 			},
 			levelLabels,
 		),
-		levelDuration: prometheus.NewCounterVec(
-			prometheus.CounterOpts{
+		levelDuration: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
 				Namespace: namespace,
 				Name:      "duration",
 				Help:      "amount of time (in ns) spent in compaction by level",

--- a/database/meterdb/db.go
+++ b/database/meterdb/db.go
@@ -92,7 +92,7 @@ type Database struct {
 	db database.Database
 
 	calls    *prometheus.CounterVec
-	duration *prometheus.CounterVec
+	duration *prometheus.GaugeVec
 	size     *prometheus.CounterVec
 }
 
@@ -112,8 +112,8 @@ func New(
 			},
 			methodLabels,
 		),
-		duration: prometheus.NewCounterVec(
-			prometheus.CounterOpts{
+		duration: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
 				Namespace: namespace,
 				Name:      "duration",
 				Help:      "time spent in database calls (ns)",

--- a/message/messages.go
+++ b/message/messages.go
@@ -142,7 +142,7 @@ type msgBuilder struct {
 
 	zstdCompressor compression.Compressor
 	count          *prometheus.CounterVec // type + op + direction
-	duration       *prometheus.CounterVec // type + op + direction
+	duration       *prometheus.GaugeVec   // type + op + direction
 
 	maxMessageTimeout time.Duration
 }
@@ -170,8 +170,8 @@ func newMsgBuilder(
 			},
 			metricLabels,
 		),
-		duration: prometheus.NewCounterVec(
-			prometheus.CounterOpts{
+		duration: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
 				Namespace: namespace,
 				Name:      "compressed_duration",
 				Help:      "time spent handling compressed messages",

--- a/network/p2p/network.go
+++ b/network/p2p/network.go
@@ -63,8 +63,8 @@ func NewNetwork(
 	namespace string,
 ) (*Network, error) {
 	metrics := metrics{
-		msgTime: prometheus.NewCounterVec(
-			prometheus.CounterOpts{
+		msgTime: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
 				Namespace: namespace,
 				Name:      "msg_time",
 				Help:      "message handling time (ns)",

--- a/network/p2p/router.go
+++ b/network/p2p/router.go
@@ -45,7 +45,7 @@ type meteredHandler struct {
 }
 
 type metrics struct {
-	msgTime  *prometheus.CounterVec
+	msgTime  *prometheus.GaugeVec
 	msgCount *prometheus.CounterVec
 }
 

--- a/snow/networking/handler/metrics.go
+++ b/snow/networking/handler/metrics.go
@@ -12,8 +12,8 @@ import (
 type metrics struct {
 	expired             *prometheus.CounterVec // op
 	messages            *prometheus.CounterVec // op
-	lockingTime         prometheus.Counter
-	messageHandlingTime *prometheus.CounterVec // op
+	lockingTime         prometheus.Gauge
+	messageHandlingTime *prometheus.GaugeVec // op
 }
 
 func newMetrics(namespace string, reg prometheus.Registerer) (*metrics, error) {
@@ -34,15 +34,15 @@ func newMetrics(namespace string, reg prometheus.Registerer) (*metrics, error) {
 			},
 			opLabels,
 		),
-		messageHandlingTime: prometheus.NewCounterVec(
-			prometheus.CounterOpts{
+		messageHandlingTime: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
 				Namespace: namespace,
 				Name:      "message_handling_time",
 				Help:      "time spent handling messages",
 			},
 			opLabels,
 		),
-		lockingTime: prometheus.NewCounter(prometheus.CounterOpts{
+		lockingTime: prometheus.NewGauge(prometheus.GaugeOpts{
 			Namespace: namespace,
 			Name:      "locking_time",
 			Help:      "time spent acquiring the context lock",

--- a/snow/networking/timeout/metrics.go
+++ b/snow/networking/timeout/metrics.go
@@ -62,7 +62,7 @@ func (m *metrics) Observe(chainID ids.ID, op message.Op, latency time.Duration) 
 // chainMetrics contains message response time metrics for a chain
 type chainMetrics struct {
 	messages         *prometheus.CounterVec // op
-	messageLatencies *prometheus.CounterVec // op
+	messageLatencies *prometheus.GaugeVec   // op
 }
 
 func newChainMetrics(reg prometheus.Registerer) (*chainMetrics, error) {
@@ -75,8 +75,8 @@ func newChainMetrics(reg prometheus.Registerer) (*chainMetrics, error) {
 			},
 			opLabels,
 		),
-		messageLatencies: prometheus.NewCounterVec(
-			prometheus.CounterOpts{
+		messageLatencies: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
 				Namespace: responseNamespace,
 				Name:      "message_latencies",
 				Help:      "message latencies (ns)",

